static/search.js

```diff
@@ -0,0 +1,72 @@
+function debounce(func, timeout = 200) {
+    let timer;
+    return (...args) => {
+        clearTimeout(timer);
+        timer = setTimeout(() => {
+            func.apply(this, args);
+        }, timeout);
+    };
+}
+
+let boxUpper = document.getElementById('searchBoxUpper');
+let boxLower = document.getElementById('searchBoxLower');
+let postElements = document.getElementsByClassName('post-item');
+let list = [];
+
+(async () => {
+    let response = await fetch('./api/v1/meta')
+    list = await response.json()
+    
+    const options = {
+        includeScore: true,
+        threshold: 0.4,
+        keys: ['title', 'author', 'text']
+    }
+    
+    const fuse = new Fuse(list, options)
+    
+    const runSearch = (() => {
+        let result = [];
+        
+        console.log("new")
+        if (boxUpper.value.length > 0) {
+            result = fuse.search(boxUpper.value)
+            for (let element of postElements) {
+                element.style.display = "none"; 
+            }
+
+            for (let item of result) {
+                for (let element of document.getElementsByClassName(`post-url-${item.item.url}`)) {
+                    element.style.display = "";
+                }
+            }
+        } else if (boxLower != null && boxLower.value.length > 0) {
+            result = fuse.search(boxLower.value)
+            for (let element of postElements) {
+                element.style.display = "none"; 
+            }
+
+            for (let item of result) {
+                for (let element of document.getElementsByClassName(`post-url-${item.item.url}`)) {
+                    element.style.display = "";
+                }
+            }
```

Comment:
seems like you can abstract some of this away since this logic is shared. for example, by saving the string value in one variable. something like:
```javascript
const value = boxUpper.value || (boxLower !== null ? boxLower.value : '');

if (value) {
  // search
} else {
  // reset display value
}
```

Reply:
Ooh this is perfect. Yeah, I should probably dive into how if statements fully work on JS (and in general).